presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -767,108 +786,256 @@ public static String keyFromPath(Path path)
+                    log.debug("HTTP endpoint detected: %s - will disable checksum validation", endpoint);
+                }
+            }
+            catch (IllegalArgumentException e) {
+                log.error("Invalid S3 endpoint URL: %s", endpoint);
+                throw new RuntimeException("Invalid S3 endpoint configuration", e);
+            }
         }
 
+        // **FIX: Build S3Configuration with checksum validation disabled for HTTP endpoints**
+        // This prevents the x-amz-content-sha256 mismatch error when using HTTP endpoints
+        // See: https://github.com/aws/aws-sdk-java-v2/issues/5498
+        final boolean disableChecksums = isHttpEndpoint;
+        S3Configuration s3Configuration = S3Configuration.builder()
+                .checksumValidationEnabled(!disableChecksums)
+                .build();
+
+        Optional<Keyring> keyring = createClientSideEncryptionKeyring(hadoopConfig);
+        if (keyring.isPresent()) {
+            log.debug("Creating S3 client with client-side encryption");
+            return createS3EncryptionClient(hadoopConfig, httpClientBuilder.build(),
+                    s3Configuration, keyring.get());
+        }
+
+        S3ClientBuilder clientBuilder = S3Client.builder()
+                .credentialsProvider(credentialsProvider)
+                .httpClient(httpClientBuilder.build())
+                .serviceConfiguration(s3Configuration)
+                .overrideConfiguration(builder -> {
+                    builder.retryStrategy(retryStrategy -> retryStrategy.maxAttempts(maxErrorRetries))
+                            .putAdvancedOption(USER_AGENT_PREFIX,
+                                    userAgentPrefix + " " + S3_USER_AGENT_SUFFIX);
+
+                    // Handle signer override if specified
+                    String signerType = hadoopConfig.get(S3_SIGNER_TYPE);
+                    if (signerType != null) {
+                        log.debug("Signer type configuration: %s (Note: v2 handles signers differently)", signerType);
+                    }
+                });
+
         boolean regionOrEndpointSet = false;
 
-        // use local region when running inside of EC2
+        // Use local region when running inside of EC2
         if (pinS3ClientToCurrentRegion) {
-            Region region = Regions.getCurrentRegion();
-            if (region != null) {
-                clientBuilder = clientBuilder.withRegion(region.getName());
-                regionOrEndpointSet = true;
+            try {
+                Region region = new DefaultAwsRegionProviderChain().getRegion();
+                if (region != null) {
+                    clientBuilder.region(region);
+                    regionOrEndpointSet = true;
+                    log.debug("Using region from provider chain: %s", region);
+                }
+            }
+            catch (Exception e) {
+                log.debug("Could not determine current region from provider chain: %s", e.getMessage());
```

Comment:
As I mentioned previously, we should deprecate this config and remove this code block here as well.